services/github/github-api-provider.js

```diff
@@ -103,7 +161,7 @@ class GithubApiProvider {
   // Act like request(), but tweak headers and query to avoid hitting a rate
   // limit. Inject `request` so we can pass in `cachingRequest` from
   // `request-handler.js`.
-  request(request, url, query, callback) {
+  request(request, url, options = {}, callback) {
```

Comment:
Was `query` always empty in the current GitHub badges?

Reply:
No, but before we were taking `query` and putting it into an `options` object, like this:

```js
request(request, url, query, callback) {

//bla

const options = {
  //bla
  qs: query,
  //bla
}

request(options...
```

Whereas now `options.qs` will already contain the query.

I've modified the call stack though `requestAsPromise()` and `createRequestFetcher()` appropriately. I don't think any of the actual badge code needs to change.

Reply:
Ahh yes, I gotcha.

There are a lot of layers of abstraction here! Would be nice to shave a couple of these off at some point. (Not that you're adding more. They were here to begin with.)

Reply:
Yes I found this part a bit tricky to understand while I was working on it.